[Fix #2044] Refactor: replace global platform-specific styles with local styles

foopang opened the pull request:
[Fix #2044] Refactor: replace global platform-specific styles with local styles

status: ready

foopang committed:
Refactor: replace global platform-specific styles with local styles

jeluard commented:
Thanks!

Please note that we don't use inline styles but define them in an associated `styles.cljs` in the same ns (if it doesn't exist please create it).
Also for `android` and `ios` specific styles we have the `defstyle` macro so you shouldn't have to use `platform-specific-style` directly. See existing usages for examples.

foopang commented:
Updated :)

jeluard commented on src/status_im/chat/views/actions.cljs:
```diff
@@ -112,17 +113,20 @@
+            public?                 [:chat :public?]
+            members                 [:current-chat-contacts]
+            status-bar-height       (get platform/platform-specific :status-bar-default-height)
+            actions-list-view-style (when platform/ios?
```
Can this be extracted in `styles` too?

jeluard commented on src/status_im/chat/views/toolbar_content.cljs:
```diff
@@ -52,8 +52,7 @@
       [view {:flex-direction :row}
-       [text {:style (get-in platform-specific [:component-styles :toolbar-last-activity])}
-        (label :t/public-group-status)]]
+       [text (label :t/public-group-status)]]
```
No more style?

foopang replied:
It was not defined in old platform-specific styles, so I think it can be safely removed here.

jeluard replied:
👍

jeluard commented on src/status_im/ui/components/chat_icon/styles.cljs:
```diff
@@ -257,7 +256,5 @@
-         {:color            :white
-          :background-color :transparent}))
-
+  {:color            :white
```
Merge is gone?

foopang replied:
The platform-specific style does not exist at all.

jeluard replied:
👍

jeluard commented on src/status_im/ui/components/status_bar/styles.cljs:
```diff
@@ -0,0 +1,101 @@
+
+(def view-transparent
+  {:background-color styles/color-transparent
+   :height           (get platform/platform-specific :status-bar-default-height)})
```
Can `status-bar-default-height` be defined in this file somehow?

foopang replied:
You mean using `defstyle`? But the value is being used in some other places not as the `height` property. Should I need to create a new `defstyle` for each different property with that value?

jeluard replied:
👍

jeluard reviewed (commented):
Great work! I believe there is still a couple places where styles can be extracted but this is definitively close to it.

jeluard commented on src/status_im/ios/platform.cljs:
```diff
@@ -114,4 +42,5 @@
    :discover                     {:uppercase-subtitles? true}
    :public-group-icon-container  {:margin-top 2}
-   :private-group-icon-container {:margin-top 2}})
+   :private-group-icon-container {:margin-top 2}
```
Can those 2 ones be moved to `styles` ?

foopang replied:
Oh, it turns out both are unused, so I just removed them.

jeluard commented on src/status_im/ui/components/status_bar/styles.cljs:
```diff
@@ -0,0 +1,101 @@
+
+(defstyle view-model-wallet
+  {:ios     {:background-color styles/color-blue4
+             :height           (get platform/platform-specific :status-bar-default-height)}
```
There are multiple calls to `(get platform/platform-specific :status-bar-default-height)` in this ns. Can you introduce a local var?

foopang commented:
@jeluard  In the latest commit, I removed the unused platform-specific properties `:public-group-icon-container` and `:private-group-icon-container`. Also cleaned up the code for status bar styles. Hope it's all good now. https://github.com/status-im/status-react/pull/2740/commits/a57527ad9e362efbcb449fd49eac54cbb63b9158

jeluard commented:
@foopang Looks good! Can you please squash your commits?

pablanopete commented:
@foopang thanks for making a contribution to Status! Can you ping me on Riot https://chat.status.im/#/register @hutch so I can personally thank ya :) 🎉

foopang commented:
@jeluard Done.

@pablanopete Thanks! I will get in touch with you soon.

asemiankevich commented:
1. Missing `add contact` and `back` buttons both Android and iOS 
@foopang @jeluard 

Branch: foopang:fix/issue-2044
Android: https://i.diawi.com/yw9RDv
iOS: https://i.diawi.com/NCdhd9

![image](https://user-images.githubusercontent.com/28752642/34214645-2382cf52-e5b4-11e7-9d30-745660dfa5c5.png)

foopang commented:
@asemiankevich sorry, didn't spot it. Fixed now.

![screenshot from 2017-12-21 12-23-08](https://user-images.githubusercontent.com/4666757/34240575-30f75658-e64a-11e7-8898-4060120a2ab7.png)

asemiankevich commented:
@foopang @jeluard App becomes irresponsive when open Wallet main screen
![image](https://user-images.githubusercontent.com/28752642/34248789-55d0eb12-e648-11e7-80dd-120addd74878.png)
TF: https://app.testfairy.com/projects/4803590-status/builds/7344703/sessions/2/?accessToken=fS6Ga1OD6R7hvf2tgUUpE2024zY
TF: https://app.testfairy.com/projects/4803622-status/builds/7344704/sessions/2/?accessToken=BiaS3566IASfHZ1KxcTrgkz0H6I

Steps:
1. Open Status
2. Set up account and /faucet some ETH
3. Open Wallet 

`0.9.11d465 (2021.1)`, `0.9.10-465-g36bcdc32+ (2022)`
----------------
Branch: foopang:fix/issue-2044
Android: https://i.diawi.com/4Z2ynx
iOS: https://i.diawi.com/NueiLJ

jeluard commented:
@asemiankevich I believe I have the same issue on `develop` so it's probably not related.

asemiankevich commented:
@jeluard I need to double check but it was working in yesterday develop build

Adding: you are right :(

foopang commented:
@asemiankevich I rebased `develop` branch so if `develop` happens to have issues would cause the same issues here too.
